controller_manager/src/controller_manager.cpp

```diff
@@ -95,20 +96,39 @@ void ControllerManager::update(const ros::Time& time, const ros::Duration& perio
   // there are controllers to start/stop
   if (please_switch_)
   {
-    // switch hardware interfaces (if any)
-    robot_hw_->doSwitch(switch_start_list_, switch_stop_list_);
+    if(!switch_started_)
+    {
+      // switch hardware interfaces (if any)
+      robot_hw_->doSwitch(switch_start_list_, switch_stop_list_);
+      switch_started_ = true;
+    }
 
     // stop controllers
-    for (unsigned int i=0; i<stop_request_.size(); i++)
-      if (!stop_request_[i]->stopRequest(time))
-        ROS_FATAL("Failed to stop controller in realtime loop. This should never happen.");
+    for (unsigned int i = 0; i < stop_request_.size(); i++)
```

Comment:
These could be converted into range-based for loops

...thanks for adding the curly braces :)

Reply:
Done!